### ISSUE TITLE
OG images

### ISSUE THREAD

**Issue by madrilene** [unlockopen/www.unlockopen.com-v4#14]
automatically create OG images for the different content types, 
- articles
- speaking
- general fallback

pulling in title, date or a custom string. 
Color coded, and with a custom SVG/Photo where available, else a coherent fallback.

**Comment by tobie** [comment]
Can we estimate the time this takes?

**Comment by madrilene** [comment]
Depends on the complexity of the design. I'd do that first in Figma so we can agree on a version for each content type. 
For the design I calculate an hour. 
The basic functionality is already there (for articles), I'd have to adjust the current template and add more for other content types.

**Comment by tobie** [comment]
Do we agree that it makes design sense to include the specific visuals (e.g. hero image, placeholder, etc.) in those?

**Comment by madrilene** [comment]
Yes! I already created a first basic version to get an idea. This just takes in the content type color scheme, title and date. Could be an MVP version for now, and enhance later. 
![Article](https://github.com/unlockopen/www.unlockopen.com-v4/assets/62096871/a2ac429a-1a1c-4add-9e53-ab09b46238b9)

**Comment by tobie** [comment]
I had seen this. My question was wrt including a dedicated image if present (e.g. hero image, video placeholder, etc.)

**Comment by madrilene** [comment]
I really like the title to be visible on those images. 
So I'd go for a version where the featured image / fallback is visible in a section, like I did for DX:

![webinar-dora-metrics-actionable (1)](https://github.com/unlockopen/www.unlockopen.com-v4/assets/62096871/dc9f9ce9-8efa-43c5-b56c-2563c17e4579)

**Comment by tobie** [comment]
Yup. That totally works for me.

**Comment by madrilene** [comment]
We have to decide on a design to be implemented: https://www.figma.com/design/CIXElK1qmSorPlNk0UPRWr/Unlock-Open-Style-Guide?node-id=456-225&t=6aPddQSJ1E5lQBAf-0

**Comment by madrilene** [comment]
![OG artivle v4](https://github.com/unlockopen/www.unlockopen.com-v4/assets/62096871/2f4db328-1415-4863-a57e-59a0a7423f78)
![OG artivle v2](https://github.com/unlockopen/www.unlockopen.com-v4/assets/62096871/2bcc5e5d-edb2-45c8-9495-fc2d904ec044)

I start implementing these versions. Any preference for the read more button?

**Comment by madrilene** [comment]
# Chosen items

## Articles
![OG artivle v4](https://github.com/unlockopen/www.unlockopen.com-v4/assets/62096871/c33c3477-b82a-4b86-81c4-1ff649ef1f9f)

## Presentations
![OG presentation v4](https://github.com/unlockopen/www.unlockopen.com-v4/assets/62096871/af0375ed-44c5-465a-9999-8e6bb6763096)

## Videos

TBD

**Comment by madrilene** [comment]
> Automatically create OGP images for the different content types.
> 
> * [x]  For **articles**, use the hero image specified in the front-matter if available, use a default background otherwise.
> * [ ]  For **presentations**, use the first page of the presentation
> * [ ]  For **videos**, use the placeholder image
> * [ ]  Fallback for rest of website
> 
> pulling in title, date or a custom string. Color coded, and with a custom SVG/Photo where available, else a coherent fallback.

We might have to rephrase this, now that we want to have a clean square image for videos and presentations as well. 
I can start with the presentations OGP image, but we should decide first where to store the (square) image that is being used. 

Also, article OGP images don't fallback, but require a dedicated image.

As for the universal fallback, I made two very simple variations. Let me know if we implement one of them first, or if I should go over it again. 

![OG universal plain](https://github.com/unlockopen/www.unlockopen.com-v4/assets/62096871/38f72bc5-2ea3-4b78-9b94-79e596feed65)
![OG universal plain white](https://github.com/unlockopen/www.unlockopen.com-v4/assets/62096871/06c3cc38-351e-401b-91b1-30917674bb32)

**Comment by tobie** [comment]
I'm not sure I understand what the question is.

**Comment by madrilene** [comment]
I mean: we have to rephrase the requirements, if you are okay with it.
**use a default background otherwise:** we have no fallback image / default for articles, because every article should always have its image assigned. 
**For presentations, use the first page of the presentation:** The idea is to use a clean image, not the first image (which contains text and is not swuare)
**For videos, use the placeholder image**: same - we need clean, square images here

Then I also shared two simple fallback images for all other contents, and wanted to know if that works for you, or you want me to iterate on that and come up with another idea.

**Comment by tobie** [comment]
Can you translate what you're saying into concrete requirements here?

I thought we had agreed on: we need a empty square picture for everything.

Let's discuss the fallback case separately. As it's a little more complicated than this imho.

**Comment by madrilene** [comment]
Exactly! :) 
Sorry if I'm not making myself clear. I just want to **adjust the written requirements**, so I can tick them off correctly. Using "first page of the presentation" or "placeholder image" is not correct anymore. 

And then we need a new issue where we can plan out how to structure the speaking post types and its respective images.

**Comment by tobie** [comment]
Ok. That makes sense. Feel free to edit and close.

**Comment by madrilene** [comment]
Universal fallback PR: https://github.com/unlockopen/www.unlockopen.com-v4/pull/39

**Comment by madrilene** [comment]
OG images for presentations and videos currently blocked by making decisions here: https://github.com/unlockopen/www.unlockopen.com-v4/issues/10
